tests/test_toolbar_monkeypatch.py

```diff
@@ -0,0 +1,111 @@
+from django.test import RequestFactory
+
+from cms.test_utils.testcases import CMSTestCase
+from cms.toolbar.toolbar import CMSToolbar
+from cms.toolbar.utils import get_object_preview_url
+
+from djangocms_versioning.cms_toolbars import VersioningToolbar
+from djangocms_versioning.test_utils.factories import (
+    PageVersionFactory,
+    UserFactory,
+)
+
+from djangocms_version_locking.models import VersionLock
+
+
+class VersionToolbarOverrideTestCase(CMSTestCase):
+
+    def _get_toolbar(self, content_obj, user, **kwargs):
+        """Helper method to set up the toolbar
+        """
+        request = RequestFactory().get(get_object_preview_url(content_obj))
+        request.user = user
+        request.session = {}
+        request.current_page = None
+        cms_toolbar = CMSToolbar(request)
+        toolbar = VersioningToolbar(
+            request,
+            toolbar=cms_toolbar,
+            is_current_app=True,
+            app_path='/',
+        )
+        toolbar.toolbar.obj = content_obj
+
+        if kwargs.get('edit_mode', False):
+            toolbar.toolbar.edit_mode_active = True
+            toolbar.toolbar.content_mode_active = False
+            toolbar.toolbar.structure_mode_active = False
+        elif kwargs.get('content_mode', False):
+            toolbar.toolbar.edit_mode_active = False
+            toolbar.toolbar.content_mode_active = True
+            toolbar.toolbar.structure_mode_active = False
+
+        request.toolbar = toolbar.toolbar
+        return toolbar
+
+    def test_not_render_edit_button_when_not_content_mode(self):
+        user = self.get_superuser()
+        version = PageVersionFactory()
+        VersionLock.objects.create(
+            version=version,
+            created_by=user,
+        )
+
+        toolbar = self._get_toolbar(version.content, user, edit_mode=True)
+        toolbar.post_template_populate()
+        buttons = toolbar.toolbar.get_right_items()[0].buttons
+        self.assertEqual(len(buttons), 1)
+        self.assertNotEqual(buttons[0].name, 'Edit')
```

Comment:
I wonder if an assert in a list would be better than targeting the first position. It's possible that when we make changes in the core the Edit button might not always be the first item in the list. I guess this would go for how we find it to delete it too.